netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -211,6 +235,11 @@ public void shutdown() {
           = Status.UNAVAILABLE.withDescription("Channel requested transport to shut down");
       handler.getWriteQueue().enqueue(new GracefulCloseCommand(status), true);
     }
+    if (keepAliveManager != null) {
+      keepAliveManager.onTransportShutdown();
```

Comment:
This should be done in the handler because:
 1. Reduce the number of different threads we do things on; this is on some random application thread
 2. This is not the only way the transport can be shutdown. For instance, if the TCP connection dies the transport will shut itself down
 3. The method is poorly named; we want to call it when the transport is terminated. It will stop doing PINGs, but if there is a long-lived stream on this transport we want to continue PINGs until it completes (and the transport becomes terminated).

channelInactive is the right event to listen for.